helper/dynamic_property_support.go

```diff
@@ -0,0 +1,85 @@
+package helper
+
+import (
+	"fmt"
+	"github.com/buildpacks/libcnb"
+	"github.com/magiconair/properties"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"os"
+	"strings"
+)
+
+const prefix = "BPL_TOMCAT_ENV_"
+
+type DynamicPropertySupport struct {
+	Bindings libcnb.Bindings
+	Logger   bard.Logger
+}
+
+func (a DynamicPropertySupport) Execute() (map[string]string, error) {
+	systemProperties := make(map[string]string)
+
+	for _, b := range a.Bindings {
+		a.Logger.Infof("Configuring properties for binding %s", b.Name)
+
+		for k, v := range b.Secret {
+			systemProperties[fmt.Sprintf("%s.%s", b.Name, k)] = v
+		}
+	}
+
+	// lets get all environment variables named BPL_TOMCAT_ENV_*
+	envVariables := os.Environ()
+	for _, envVariable := range envVariables {
+		parts := strings.Split(envVariable, "=")
+		if strings.HasPrefix(parts[0], prefix) {
+			a.Logger.Infof("Configuring property from env var %s", parts[0])
+			value := os.Getenv(parts[0])
+			systemProperties[toSystemProperty(parts[0])] = value
+		}
+	}
+
+	catalinaProperties := os.Getenv("BPI_CATALINA_CONFIG")
```

Comment:
There's a helper for this type of thing that we prefer to use: `sherpa.GetEnvRequired`. It returns a reasonably formatted error, so you can just `if err != nil { return nil, err }`.

https://github.com/paketo-buildpacks/libpak/blob/main/sherpa/env_var.go#L39